fedimintd/src/fedimintd.rs

```diff
@@ -205,7 +218,10 @@ impl Fedimintd {
             bitcoind_rpc,
             server_gens: ServerModuleInitRegistry::new(),
             server_gen_params: ServerModuleConfigGenParamsRegistry::default(),
-            version_hash: version_hash.to_owned(),
+            code_version_hash: code_version_hash.to_owned(),
+            code_version_str: code_version_vendor_suffix
```

Comment:
Instead of having this awkward `vendor_suffix`, why not serialize the available module kinds in ascending order? That should achieve 95% of what we want here and won't require an API change, so we could even backport.

Reply:
@elsirion 

Hmm... That would probably work, but conceptually... is it only about module kinds?

If there's a vendor xyz, and it provides some tweaked `fedimintd`, wouldn't we want it to enforce that all guardians run the version from the same vendor?

The `vendor_suffix` is also useful in other places. If I run `fedimintd --version` etc. I'd like to see `fedimint (v0.4.5-fedi)` etc. So I don't think I'd want to get rid of "awkward vendor_suffix" anyway.

Reply:
I'm ok going with this version, but you kinda have to trust vendors to actually set this. Maybe we could combine both?

Reply:
Feel free to merge or squash in 5e85052b37dbfbe901e8ea05c21fd6fc04e64061